### PR TITLE
Use doc comments over help text with `StructOpt`

### LINKED ISSUE #1852: Move structopt help to doc comments
**Issue summary**

We should specify all the `help` messages in `structopt` using doc comments instead. This would bring us spell-checking as well as documentation for free.

ie [here](https://github.com/ChainSafe/forest/blob/main/forest/src/cli/mod.rs#L115).

Search codebase for all `help =` occurences.

**Other information and links**

n/a

### PR DESCRIPTION
**Pull request by connormullett**
**Summary of changes**
Changes introduced in this pull request:
- Use doc comments over help text to enable spell checking StructOpt help text

**Reference issue to close (if applicable)**

Closes #1852 

**Other information and links**

### CODE REVIEW

File: forest/src/cli/chain_cmd.rs [hunk 1]
```diff
@@ -50,26 +51,28 @@ pub enum ChainCommands {
     },
 
     /// Prints out the genesis tipset
-    #[structopt(about = "Prints genesis tipset", help = "Prints genesis tipset")]
+    #[structopt(about = "Prints genesis tipset")]
     Genesis,
 
     /// Prints out the canonical head of the chain
-    #[structopt(about = "Print chain head", help = "Print chain head")]
+    #[structopt(about = "Print chain head")]
```
Comment (LesnyRumcajs) [inline forest/src/cli/chain_cmd.rs:58]: In general, do we need those `about`s? To my understanding, they apply only to the top-level app.

File: forest/src/cli/chain_cmd.rs [hunk 2]
```diff
@@ -21,7 +21,8 @@ pub enum ChainCommands {
     /// Retrieves and prints out the block specified by the given CID
     #[structopt(about = "<Cid> Retrieve a block and print its details")]
     Block {
-        #[structopt(short, help = "Input a valid CID")]
+        #[structopt(short)]
+        /// Input a valid CID
```
Comment (LesnyRumcajs) [inline forest/src/cli/chain_cmd.rs:25]: Let's use the same pattern everywhere, i.e. doc comment first, then structopt and field definition.
```suggestion
        /// Input a valid CID
        #[structopt(short)]
```

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
Nice change!

Let's use the same pattern everywhere, i.e. doc comment first, then structopt and field definition. It would also be great to get rid of `about` the same way (unless it's not possible).

File: forest/src/cli/mpool_cmd.rs [hunk 3]
```diff
@@ -22,17 +22,15 @@ use crate::cli::handle_rpc_err;
...
-            short,
-            help = "Number of blocks to lookback for minimum base fee",
-            default_value = "60"
-        )]
+        /// Number of blocks to look back for minimum base fee
+        #[structopt(short, default_value = "60")]
         base_fee_lookback: u32,
-        #[structopt(short, help = "Print stats for local addresses only")]
+        #[structopt(short)]
+        /// Print stats for local addresses only
```
Comment (elmattic) [inline forest/src/cli/mpool_cmd.rs:33]: Let's put doc comment always on top as Hubert suggested.

File: forest/src/cli/chain_cmd.rs (hunk 1 above)
Comment (elmattic) [reply forest/src/cli/chain_cmd.rs:58]: We don't need the abouts if we already have doc comments.

```
/// Prints out the genesis tipset
#[structopt(about = "Prints genesis tipset")]
Genesis,
```
So in this case it's redundant. Let's keep the more up to date comment/about and put it in a doc comment for all `#[structopt(about =`.

File: forest/src/cli/mpool_cmd.rs (hunk 3 above)
Comment (connormullett) [reply forest/src/cli/mpool_cmd.rs:32]: Should be all fixed

File: forest/src/cli/auth_cmd.rs [hunk 4]
```diff
@@ -12,22 +12,15 @@ use forest_auth::*;
...
-    #[structopt(about = "Get RPC API information")]
+    /// Get RPC API Information
     ApiInfo {
-        #[structopt(
-            short,
-            long,
-            help = "permission to assign the token, one of: read, write, sign, admin"
-        )]
+        #[structopt(short, long)]
+        /// permission to assign the token, one of: read, write, sign, admin
```
Comment (LesnyRumcajs) [inline forest/src/cli/auth_cmd.rs:23]:
```suggestion
        /// permission to assign the token, one of: read, write, sign, admin
        #[structopt(short, long)]
```

File: forest/src/cli/chain_cmd.rs [hunk 5]
```diff
@@ -50,26 +49,24 @@ pub enum ChainCommands {
...
         cid: String,
     },
 
     /// Reads and prints out IPLD nodes referenced by the specified CID from chain
     /// block store and returns raw bytes
-    #[structopt(about = "<CID> Read the raw bytes of an object")]
     ReadObj {
-        #[structopt(short, help = "Input a valid CID")]
+        #[structopt(short)]
+        /// Input a valid CID
```
Comment (LesnyRumcajs) [inline forest/src/cli/chain_cmd.rs:69]:
```suggestion
        /// Input a valid CID
        #[structopt(short)]
```

File: forest/src/cli/fetch_params_cmd.rs [hunk 6]
```diff
@@ -12,13 +12,17 @@ use super::cli_error_and_die;
 #[allow(missing_docs)]
 #[derive(Debug, StructOpt)]
 pub struct FetchCommands {
-    #[structopt(short, long, help = "Download all proof parameters")]
+    #[structopt(short, long)]
+    /// Download all proof parameters
```
Comment (LesnyRumcajs) [inline forest/src/cli/fetch_params_cmd.rs:16]:
```suggestion
    /// Download all proof parameters
    #[structopt(short, long)]
```

File: forest/src/cli/fetch_params_cmd.rs [hunk 7]
```diff
@@ -12,13 +12,17 @@ use super::cli_error_and_die;
...
     all: bool,
-    #[structopt(short, long, help = "Download only verification keys")]
+    #[structopt(short, long)]
+    /// Download only verification keys
```
Comment (LesnyRumcajs) [inline forest/src/cli/fetch_params_cmd.rs:19]:
```suggestion
    /// Download only verification keys
    #[structopt(short, long)]
```

File: forest/src/cli/fetch_params_cmd.rs [hunk 8]
```diff
@@ -12,13 +12,17 @@ use super::cli_error_and_die;
...
     keys: bool,
-    #[structopt(required_ifs(&[("all", "false"), ("keys", "false")]), help = "Size in bytes")]
+    #[structopt(required_ifs(&[("all", "false"), ("keys", "false")]))]
+    /// Size in bytes
```
Comment (LesnyRumcajs) [inline forest/src/cli/fetch_params_cmd.rs:22]:
```suggestion
    /// Size in bytes
    #[structopt(required_ifs(&[("all", "false"), ("keys", "false")]))]
```

File: forest/src/cli/fetch_params_cmd.rs [hunk 9]
```diff
@@ -12,13 +12,17 @@ use super::cli_error_and_die;
...
     params_size: Option<String>,
-    #[structopt(short, long, help = "Show verbose logging")]
+    #[structopt(short, long)]
+    /// Show verbose logging
```
Comment (LesnyRumcajs) [inline forest/src/cli/fetch_params_cmd.rs:25]:
```suggestion
    /// Show verbose logging
    #[structopt(short, long)]
```

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: forest/src/cli/auth_cmd.rs (hunk 4 above)
Comment (connormullett) [reply forest/src/cli/auth_cmd.rs:23]: 👍

File: forest/src/cli/chain_cmd.rs (hunk 5 above)
Comment (connormullett) [reply forest/src/cli/chain_cmd.rs:69]: 👍

File: forest/src/cli/fetch_params_cmd.rs (hunk 6 above)
Comment (connormullett) [reply forest/src/cli/fetch_params_cmd.rs:16]: 👍

File: forest/src/cli/fetch_params_cmd.rs (hunk 7 above)
Comment (connormullett) [reply forest/src/cli/fetch_params_cmd.rs:19]: 👍

File: forest/src/cli/fetch_params_cmd.rs (hunk 8 above)
Comment (connormullett) [reply forest/src/cli/fetch_params_cmd.rs:22]: 👍

File: forest/src/cli/fetch_params_cmd.rs (hunk 9 above)
Comment (connormullett) [reply forest/src/cli/fetch_params_cmd.rs:25]: 👍

**Review by LesnyRumcajs — APPROVED** [review]
Nice!

File: forest/src/cli/net_cmd.rs [hunk 10]
```diff
@@ -14,21 +14,17 @@ use forest_rpc_client::net_ops::*;
...
     Listen,
     /// Lists `libp2p` swarm peers
-    #[structopt(about = "Print peers")]
     Peers,
-    /// Connects to a peer
-    #[structopt(about = "Connect to a peer by its peer ID and multiaddresses")]
+    /// Connects to a peer by its peer ID and multi-addresses
     Connect {
-        #[structopt(about = "Multiaddr (with /p2p/ protocol)")]
+        /// Multi-address (with /P2P/ protocol)
```
Comment (elmattic) [inline forest/src/cli/net_cmd.rs:22]: I would use more `/p2p/` here (so in quotes ``).

**Review by elmattic — CHANGES_REQUESTED** [review]

File: forest/src/cli/mod.rs [hunk 11]
```diff
@@ -80,107 +80,107 @@ pub struct Cli {
...
+    /// Encrypt the key-store (default: true)
+    #[structopt(long)]
     pub encrypt_keystore: Option<bool>,
     #[structopt(
         long,
-        help = "Choose network chain to sync to",
         default_value = "mainnet",
         possible_values = &["mainnet", "calibnet"],
     )]
+    /// Choose network chain to sync to
```
[the remark this answers is not in the document]
Comment (connormullett) [reply forest/src/cli/mod.rs:183]: Good catch!

File: forest/src/cli/net_cmd.rs (hunk 10 above)
Comment (connormullett) [reply forest/src/cli/net_cmd.rs:22]: 👍

**Review by elmattic — APPROVED** [review]

**Pull request merged**